assets/js/utils.js

```diff
@@ -0,0 +1,45 @@
+const utils = (()=>{
+
+  // utils.date
+  function date(format){
+    let today = new Date();
+    const dd = today.getDate();
+    const mm = today.getMonth() + 1; //January is 0!
+  
+    var yyyy = today.getFullYear();
+    if (dd < 10) dd = '0' + dd;
+    if (mm < 10) mm = '0' + mm;
+  
+    //today =  `${yyyy}-${mm}-${dd}`;
+    format = format ? format : 'mm-dd-yyyy';
+    format = format.replace('dd',dd).replace('mm',mm).replace('yyyy',yyyy);
+    return format;
+  }
+  // utils.time
+  function time(){
+    let today = new Date();
+    let hh = today.getHours();
+    hh = hh.toString().length === 1 ? `0${hh}` : hh
+    let mm = today.getMinutes();
+    mm = mm.toString().length === 1 ? `${mm}0` : mm
+
+    return `${hh}:${mm}`;
+  }
+  // utils.datetime
+  function datetime(){
+      return `${date()} ${time()}`
+  }
+  // utils.uid
+  function uid(){  // generate unique id
+      const s4 = () => Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
```

Comment:
are you sure this is ALWAYS unique???

Reply:
Not really sure. See https://stackoverflow.com/questions/38228180/is-generating-and-concatenating-3-math-random-values-more-random-than-1-math-r

Reply:
Added uuidv4 for as better solution